jsettlers.logic/src/main/java/jsettlers/logic/map/grid/partition/manager/materials/requests/MaterialsForBuildingsRequestPriorityQueue.java

```diff
@@ -114,6 +114,9 @@ protected MaterialRequestObject getRequestForPriority(int priority) {
 	}
 
 	private int getRandomStartIndex() {
+		if (settings.drawRandomBuilding() == null){
```

Comment:
@rpolzer: Please assign settings.drawRandomBuilding() to a variable and then use that variable later in line 120. This way you don't draw two (different) random buildings. Additionally, this should fix the replay test, as then the behavior should not be different then bevore.